### PR TITLE
Disallow event tracking with empty/whitespace categories

### LINKED ISSUE #11623: Action table for type '12' was not found during Actions archiving
Hi,
I started using Piwik a week ago, tracking 2 websites. 
The archiving cron runs every hour. 

Now I recognized that the archiving stopped working for one of the two websites 4 days ago. It worked for both websites before (see the screenshot: https://drive.google.com/file/d/0B7Ss7W8DbUYtNm9rOFkzbUxFMTQ/view?usp=sharing)

Here is the full output from the archiving job:
```
root@server1:/var/www/piwik.xxxxxx.de/piwik# /var/www/piwik.jstmail.de/piwik/console core:archive --url=https://piwik.xxxxxx.de

INFO [2017-04-18 13:31:13] ---------------------------
INFO [2017-04-18 13:31:13] INIT
INFO [2017-04-18 13:31:13] Running Piwik 3.0.3 as Super User
INFO [2017-04-18 13:31:13] ---------------------------
INFO [2017-04-18 13:31:13] NOTES
INFO [2017-04-18 13:31:13] - Reports for today will be processed at most every 150 seconds. You can change this value in Piwik UI > Settings > General Settings.
INFO [2017-04-18 13:31:13] - Reports for the current week/month/year will be refreshed at most every 3600 seconds.
INFO [2017-04-18 13:31:13] - Archiving was last executed without error 4 days 2 hours ago
INFO [2017-04-18 13:31:13] - Will process 2 other websites because the last time they were archived was on a different day (in the website's timezone) , IDs: 1, 2
INFO [2017-04-18 13:31:13] - Will process 1 other websites because some old data reports have been invalidated (eg. using the Log Import script) , IDs: 2
INFO [2017-04-18 13:31:13] ---------------------------
INFO [2017-04-18 13:31:13] START
INFO [2017-04-18 13:31:13] Starting Piwik reports archiving...
INFO [2017-04-18 13:31:13] Day has finished for website id 1 since last run
INFO [2017-04-18 13:31:13] Will pre-process for website id = 1, period = day, date = last10
INFO [2017-04-18 13:31:13] - pre-processing all visits
INFO [2017-04-18 13:31:14] Archived website id = 1, period = day, 0 segments, 749 visits in last 10 days, 107 visits today, Time elapsed: 0.810s
INFO [2017-04-18 13:31:14] Will pre-process for website id = 1, period = week, date = last2
INFO [2017-04-18 13:31:14] - pre-processing all visits
INFO [2017-04-18 13:31:15] Archived website id = 1, period = week, 0 segments, 749 visits in last 2 weeks, 259 visits this week, Time elapsed: 0.628s
INFO [2017-04-18 13:31:15] Will pre-process for website id = 1, period = month, date = last2
INFO [2017-04-18 13:31:15] - pre-processing all visits
INFO [2017-04-18 13:31:15] Archived website id = 1, period = month, 0 segments, 749 visits in last 2 months, 749 visits this month, Time elapsed: 0.720s
INFO [2017-04-18 13:31:15] Will pre-process for website id = 1, period = year, date = last2
INFO [2017-04-18 13:31:15] - pre-processing all visits
INFO [2017-04-18 13:31:16] Archived website id = 1, period = year, 0 segments, 749 visits in last 2 years, 749 visits this year, Time elapsed: 0.599s
INFO [2017-04-18 13:31:16] Archived website id = 1, 4 API requests, Time elapsed: 2.763s [1/2 done]
INFO [2017-04-18 13:31:16] Day has finished for website id 2 since last run
INFO [2017-04-18 13:31:16] Will pre-process for website id = 2, period = day, date = last10
INFO [2017-04-18 13:31:16] - pre-processing all visits
ERROR [2017-04-18 13:31:17] Got invalid response from API request: ?module=API&method=API.get&idSite=2&period=day&date=last10&format=php&trigger=archivephp. Response was 'a:2:{s:6:"result";s:5:"error";s:7:"message";s:93:"Action table for type '12' was not found during Actions archiving. - caused by plugin Actions";}'
ERROR [2017-04-18 13:31:17] Empty or invalid response '' for website id 2, Time elapsed: 0.377s, skipping
INFO [2017-04-18 13:31:17] Done archiving!
INFO [2017-04-18 13:31:17] ---------------------------
INFO [2017-04-18 13:31:17] SUMMARY
INFO [2017-04-18 13:31:17] Total visits for today across archived websites: 107
INFO [2017-04-18 13:31:17] Archived today's reports for 1 websites
INFO [2017-04-18 13:31:17] Archived week/month/year for 1 websites
INFO [2017-04-18 13:31:17] Skipped 1 websites
INFO [2017-04-18 13:31:17] - 0 skipped because no new visit since the last script execution
INFO [2017-04-18 13:31:17] - 0 skipped because existing daily reports are less than 150 seconds old
INFO [2017-04-18 13:31:17] - 0 skipped because existing week/month/year periods reports are less than 3600 seconds old
INFO [2017-04-18 13:31:17] - 1 skipped because got an error while querying reporting API
INFO [2017-04-18 13:31:17] Total API requests: 4
INFO [2017-04-18 13:31:17] done: 1/2 50%, 107 vtoday, 1 wtoday, 1 wperiods, 4 req, 3341 ms, 2 errors.
INFO [2017-04-18 13:31:17] Time elapsed: 3.341s
INFO [2017-04-18 13:31:17] ---------------------------
INFO [2017-04-18 13:31:17] SCHEDULED TASKS
INFO [2017-04-18 13:31:17] Starting Scheduled tasks...
INFO [2017-04-18 13:31:17] done
INFO [2017-04-18 13:31:17] ---------------------------
INFO [2017-04-18 13:31:17] ---------------------------
INFO [2017-04-18 13:31:17] SUMMARY OF ERRORS
INFO [2017-04-18 13:31:17] Error: Got invalid response from API request: ?module=API&method=API.get&idSite=2&period=day&date=last10&format=php&trigger=archivephp. Response was 'a:2:{s:6:"result";s:5:"error";s:7:"message";s:93:"Action table for type '12' was not found during Actions archiving. - caused by plugin Actions";}'
INFO [2017-04-18 13:31:17] Error: Empty or invalid response '' for website id 2, Time elapsed: 0.377s, skipping
ERROR [2017-04-18 13:31:17] 2 total errors during this script execution, please investigate and try and fix these errors.
```

Edit: This is the system summary. System-check is all green. I did not install any additional plugins. If you need additional information please ask. :)

> 1 Benutzer
> 4 Segmente
> 2 Websites
> 43 aktivierte Plugins
> Piwik-Version: 3.0.3
> MySQL-Version: 5.5.54
> PHP-Version: 7.0.18-1~dotdeb+8.1

### PR DESCRIPTION
**Pull request by sgiehl**
fixes #11623

### CODE REVIEW

File: js/piwik.js [hunk 1]
```diff
@@ -4957,7 +4957,7 @@ if (typeof window.Piwik !== 'object') {
             function logEvent(category, action, name, value, customData, callback)
             {
                 // Category and Action are required parameters
-                if (String(category).length === 0 || String(action).length === 0) {
+                if (this.trim(String(category)).length === 0 || this.trim(String(action)).length === 0) {
```
Comment (sgiehl) [inline js/piwik.js]: This should prevent requests being sent with whitespaced category or action

File: plugins/Actions/ArchivingHelper.php [hunk 2]
```diff
@@ -53,7 +53,7 @@ public static function updateActionsTableWithRowQuery($query, $fieldQueried, & $
                 unset($row[PiwikMetrics::INDEX_SITE_SEARCH_HAS_NO_RESULT]);
             }
 
-            if (in_array($row['type'], array(Action::TYPE_CONTENT, Action::TYPE_EVENT))) {
+            if (in_array($row['type'], array(Action::TYPE_CONTENT, Action::TYPE_EVENT, Action::TYPE_EVENT_NAME))) {
```
Comment (sgiehl) [inline plugins/Actions/ArchivingHelper.php]: This should fix the failing archiving for already tracked empty categories.

File: js/piwik.js (hunk 1 above)
Comment (tsteur) [inline js/piwik.js]: Should we log an error to the console in this case?

**Comment by tsteur** [comment]
Looks good to me. To be defined if we log an error or not in tracker.

**Comment by sgiehl** [comment]
As this case seems currently to be the only case where we "actively" discard an tracking request guess logging an error wouldn't be bad. Updated the PR :)

**Comment by tsteur** [comment]
LGTM if tests pass

**Comment by sgiehl** [comment]
Build failure seems not to be related to this changes

**Pull request merged**